diff --git a/src/interface/efi/efi_driver.c b/src/interface/efi/efi_driver.c
index fd9be5f5..e6b0c491 100644
--- a/src/interface/efi/efi_driver.c
+++ b/src/interface/efi/efi_driver.c
@@ -320,7 +320,7 @@ efi_driver_stop ( EFI_DRIVER_BINDING_PROTOCOL *driver __unused,
 	if ( ! efidev ) {
 		DBGCP ( device, "EFIDRV %s is not started\n",
 			efi_handle_name ( device ) );
-		return EFI_DEVICE_ERROR;
+		return 0;
 	}
 
 	/* Raise TPL */
