diff --git a/src/interface/efi/efi_rng.c b/src/interface/efi/efi_rng.c
index b76a6fc0d..5c1f8a01b 100644
--- a/src/interface/efi/efi_rng.c
+++ b/src/interface/efi/efi_rng.c
@@ -23,6 +23,7 @@
 
 FILE_LICENCE ( GPL2_OR_LATER_OR_UBDL );
 
+#include <unistd.h>
 #include <errno.h>
 #include <ipxe/entropy.h>
 #include <ipxe/crc32.h>
@@ -53,6 +54,7 @@ EFI_REQUEST_PROTOCOL ( EFI_RNG_PROTOCOL, &efirng );
  * report the actual amount of entropy returned by GetRNG().
  */
 #define EFIRNG_LEN 32
+#define MAX_EFIRNG_RETRIES 100
 
 /**
  * Enable entropy gathering
@@ -92,8 +94,15 @@ static int efirng_get_noise ( noise_sample_t *noise ) {
 	assert ( efirng != NULL );
 
 	/* Get the minimum allowed number of random bytes */
-	if ( ( efirc = efirng->GetRNG ( efirng, NULL, sizeof ( buf ),
-					buf ) ) != 0 ) {
+	for(int retry=0; retry<MAX_EFIRNG_RETRIES; retry++) {
+		efirc = efirng->GetRNG ( efirng, NULL, sizeof ( buf ), buf );
+		if ( efirc == EFI_NOT_READY ) {
+			usleep(1000);
+		} else {
+			break;
+		}
+	}
+	if ( efirc != 0 ) {
 		rc = -EEFI ( efirc );
 		DBGC ( &efirng, "ENTROPY could not read from RNG: %s\n",
 		       strerror ( rc ) );
